### PR TITLE
backport: merge bitcoin#21732, #21762, #21754, #21953, #21850, #22633, #22738, #23154, #23721, #24002, #24197, merge bitcoin-core/gui#399 (auxiliary backports: part 14)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependency for https://github.com/dashpay/dash/pull/6138

* In [bitcoin#21754](https://github.com/bitcoin/bitcoin/pull/21754), the `scriptSig` padding multiplier (`24`) differs from upstream (`35`) as the `vsize` value it corresponds to must match what is ordinarily generated (`85` vs `96` upstream) in order to fulfill an assertion ([source](https://github.com/dashpay/dash/blob/d9835515cc1e1fd7d4fd3006b51a141fa2265613/test/functional/test_framework/wallet.py#L107)).

* In [bitcoin#21953](https://github.com/bitcoin/bitcoin/pull/21953), the hash associated with height `200` is generated like this (this is the same method used in [dash#5236](https://github.com/dashpay/dash/pull/5236)):
  * Add the height desired to the `CRegTestParams::m_assumeutxo_data` map with a garbage hash value (like `uint256::ONE`). This is to avoid an unrecognized metadata failure ([source](https://github.com/dashpay/dash/blob/5211886fb44c839d9197599d39908e1b707dfc7c/src/validation.cpp#L5755-L5761)) caused by looking through the map to see if the height's there.
  * Change the `LogPrintf(..)` in the serialized hash check error log message located [here](https://github.com/dashpay/dash/blob/5211886fb44c839d9197599d39908e1b707dfc7c/src/validation.cpp#L5876-L5880) to a `std::cout << strprintf(..)`
  * Edit the value of `mineBlocks` [here](https://github.com/dashpay/dash/blob/5211886fb44c839d9197599d39908e1b707dfc7c/src/test/validation_chainstatemanager_tests.cpp#L248-L253) to be 100 blocks _less_ than the desired height.
  * Compile Dash Core and run `./src/test/test_dash -t validation_chainstatemanager_tests`
  * Take the `got` value printed to your terminal window/`stdout` (the `expected` value should be our garbage value from earlier, ignore that). That's your good hash.
  * Update the `CRegTestParams::m_assumeutxo_data` map entry with the correct entry, reverse every change _except_ the map entry (for obvious reasons) and the `mineBlocks` change.
    * Remember to add/update the hash [here](https://github.com/dashpay/dash/blob/5211886fb44c839d9197599d39908e1b707dfc7c/src/test/validation_tests.cpp#L29-L31) in `validation_tests`, it simply tests the hardcoded chainparams value with its own hardcoded value. That's also why we don't use this test since it'll just regurgitate the garbage values we give it.
  * Compile and re-run the test. If it passes, your hash is good. Revert the `mineBlocks` change.
  * Profit?

## Breaking Changes

None expected.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commits/pr_6137/ (CI: https://gitlab.com/UdjinM6/dash/-/pipelines/1382702435)

pls note that 22501 and 397b432 are also parts of #6133, 22619 is a part of #6071

**Review by UdjinM6 — APPROVED** [review]
utACK 1c5ea38c681d82a1ae8a30a15ef377f77e9c507a

**Review by PastaPastaPasta — APPROVED** [review]
utACK 1c5ea38c681d82a1ae8a30a15ef377f77e9c507a

File: contrib/linearize/linearize-data.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
 import time
 import glob
 from collections import namedtuple
-from binascii import unhexlify
```
Comment (knst) [inline contrib/linearize/linearize-data.py:20]: missing changes in `contrib/signet/miner` - should be partial https://github.com/bitcoin/bitcoin/pull/22633

File: doc/assumeutxo.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+The RPC commands `dumptxoutset` and `loadtxoutset` are used to respectively generate
+and load UTXO snapshots. The utility script `./contrib/devtools/utxo_snapshot.sh` may
+be of use.
+
+## General background
+
+- [assumeutxo proposal](https://github.com/jamesob/assumeutxo-docs/tree/2019-04-proposal/proposal)
+- [Github issue](https://github.com/bitcoin/bitcoin/issues/15605) (Bitcoin)
+- [draft PR](https://github.com/bitcoin/bitcoin/pull/15606) (Bitcoin)
```
Comment (knst) [inline doc/assumeutxo.md:14]: nit: https://github.com/dashpay/dash-issues/issues/36 can be added here

**Pull request merged**